code/Controllers/CMSPageAddController.php

```diff
@@ -192,12 +192,7 @@ public function AddForm()
         return $form;
     }
 
-    /**
-     * @param array $data
-     * @param Form $form
-     * @return HTTPResponse
-     */
-    public function doAdd($data, $form)
+    public function doAdd(array $data, Form $form): HTTPResponse
```

Comment:
This method sometimes returns `Security::permissionFailure()` which sometimes returns an array

Reply:
As noted above, it does not return an array